src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -934,12 +914,27 @@ impl Coordinator {
             Some(path) => {
                 // For everything else, return the plan for the stage identified
                 // by the corresponding path.
-                let row = trace
-                    .into_iter()
-                    .find(|entry| entry.path == path)
+
+                // A helper struct hat removes the first (and by assumption -
```

Comment:
This smartens the logic to extract the "fast path" plan if (a) the requested optimizer stage should return the fast path plan (if present), and (b) `no_fast_path` is not set.